pkg/backup/backupdest/backup_index.go

```diff
@@ -130,3 +196,30 @@ func getBackupIndexFileName(startTime, endTime hlc.Timestamp) string {
 		descEndTs, formattedStartTime, formattedEndTime,
 	)
 }
+
+// flattenSubdirForIndex flattens a full backup subdirectory to be used in the
+// index. Note that this path does not contain a trailing or leading slash.
+// It assumes subdir is not `LATEST` and has been resolved.
+// We flatten the subdir so that when listing from the index, we can list with
+// the `index/` prefix and delimit on `/`. e.g.:
+//
+// index/
+//
+//	|_ 2025-08-13-120000.00/
+//	|  |_ <index_meta>.pb
+//	|_ 2025-08-14-120000.00/
+//	|  |_ <index_meta>.pb
+//	|_ 2025-08-14-120000.00/
+//		 |_ <index_meta>.pb
+//
+// Listing on `index/` and delimiting on `/` will return the subdirectories
+// without listing the files in them.
+func flattenSubdirForIndex(subdir string) string {
+	// Trimming any trailing and leading slashes guarantees a specific format when
+	// returning the flattened subdir, so callers can expect a consistent result.
+	flattened, _ := strings.CutSuffix(strings.ReplaceAll(
```

Comment:
i think we need to cut the suffix, and then do the replace all, correct? Or else, the trailing / will get converted to a -.

Reply:
🤦🏻 Yup